runtimevar/awssecretsmanager/awssecretsmanager.go

```diff
@@ -0,0 +1,359 @@
+// Copyright 2020 The Go Cloud Development Kit Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package awssecretsmanager provides a runtimevar implementation with variables
+// read from AWS Secrets Manager (https://aws.amazon.com/secrets-manager)
+// Use OpenVariable to construct a *runtimevar.Variable.
+//
+// URLs
+//
+// For runtimevar.OpenVariable, awssecretsmanager registers for the scheme "awssecretsmanager".
+// The default URL opener will use an AWS session with the default credentials
+// and configuration; see https://docs.aws.amazon.com/sdk-for-go/api/aws/session/
+// for more details.
+// To customize the URL opener, or for more details on the URL format,
+// see URLOpener.
+// See https://gocloud.dev/concepts/urls/ for background information.
+//
+// As
+//
+// awssecretsmanager exposes the following types for As:
+//  - Snapshot: *secretsmanager.GetSecretValueOutput, *secretsmanager.DescribeSecretOutput
+//  - Error: awserr.Error
+package awssecretsmanager // import "gocloud.dev/runtimevar/awssecretsmanager"
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"path"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/google/wire"
+	gcaws "gocloud.dev/aws"
+	"gocloud.dev/gcerrors"
+	"gocloud.dev/runtimevar"
+	"gocloud.dev/runtimevar/driver"
+)
+
+func init() {
+	runtimevar.DefaultURLMux().RegisterVariable(Scheme, new(lazySessionOpener))
+}
+
+// Set holds Wire providers for this package.
+var Set = wire.NewSet(
+	wire.Struct(new(URLOpener), "ConfigProvider"),
+)
+
+// URLOpener opens AWS Secrets Manager URLs like "awssecretsmanager://myvar".
```

Comment:
Can you add a reference to the "AWS Secrets Manager doc" that describes what "myvar" is here, and also in the docstring for OpenVariable? ("name" is not very descriptive).